src/public/modules/forms/helpers/response-factory.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  ArrayResponse,
+  DisplayedResponseWithoutAnswer,
+  NestedResponse,
+  SingleResponse,
+} from '../../../../types/response'
+
+import {
+  ArrayAnswerResponse,
+  ErrorResponse,
+  Response,
+  SingleAnswerResponse,
+  TableResponse,
+} from './csv-response-classes'
+
+export const getResponseInstance = (
+  fieldRecordData: DisplayedResponseWithoutAnswer,
+): Response => {
+  switch (fieldRecordData.fieldType) {
+    case 'table':
+      if (isNestedResponse(fieldRecordData)) {
+        return new TableResponse(fieldRecordData)
+      }
+      break
+    case 'checkbox':
+      if (isArrayResponse(fieldRecordData)) {
+        return new ArrayAnswerResponse(fieldRecordData)
+      }
+      break
+    default:
+      if (isSingleResponse(fieldRecordData)) {
+        return new SingleAnswerResponse(fieldRecordData)
+      }
+      break
+  }
+  return new ErrorResponse(fieldRecordData)
+}
+
+function isNestedResponse(
+  response: DisplayedResponseWithoutAnswer,
+): response is NestedResponse {
+  return 'answerArray' in response
```

Comment:
this is not quite enough to determine that the response is a `NestedResponse`, right? we should also check that the array is in fact nested. the `hasProp` utility in the backend might come in useful here; you can consider moving that into the `shared` folder so the frontend can make use of it.